### PR TITLE
Order gang scheduled jobs based on time of arrival

### LINKED ISSUE #454: Order Job/Task by creation time
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**Description**:

Currently, if plugin can not decide the order, kube-batch will use job/task's name to order them. IMO, it's better to use creation time; if creation time equal, use job/task's name.

### PR DESCRIPTION
**Pull request by adam-marek**
**What this PR does / why we need it**:
The ordering of equal priority gang scheduled jobs is currently based on the job's namespace/name which is unintuitive when multiple users schedule jobs competing for the same resources.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #454 

**Special notes for your reviewer**:

**Release note**:

```release-note
Equal priority gang scheduled jobs will be ordered by their time of creation.
```

### CODE REVIEW

File: pkg/scheduler/plugins/gang/gang.go [hunk 1]
```diff
@@ -120,7 +120,11 @@ func (gp *gangPlugin) OnSessionOpen(ssn *framework.Session) {
 		}
 
 		if !lReady && !rReady {
-			if lv.UID < rv.UID {
+			if lv.PodGroup.GetCreationTimestamp().Time.Equal(rv.PodGroup.GetCreationTimestamp().Time) {
```
Comment (k82cn) [inline pkg/scheduler/plugins/gang/gang.go]: hm... we need to add a new field into JobInfo, the creation-timestamp is from `PodGroup` or `PDB`. And we will check jobInfo's creation time here. We need to support PDB as tf-operator is using it, we will remove it at least after 0.4 release.

Comment (adam-marek) [reply pkg/scheduler/plugins/gang/gang.go]: Something along these lines ? (https://github.com/kubernetes-sigs/kube-batch/pull/464/commits/3f0dfd856741f6d0365962fdd6b652eb06ede2e3)

**Review by k82cn — COMMENTED** [review]
LGTM overall! let’s get it merged when CI is happy :)

**Comment by k82cn** [comment]
please sign cla

**Comment by adam-marek** [comment]
CLA should be signed now

**Pull request merged**